GeoQuiz/app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -98,35 +100,34 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun updateQuestion() {
-        val questionTextResId = questionBank[currentIndex].textResId
+        val questionTextResId = quizViewModel.currentQuestionText
         questionTextView.setText(questionTextResId)
 
-        if (questionBank[currentIndex] in answeredQuestion) {
+        if (quizViewModel.checkQuestion()) {
             disableButtons()
         } else {
             enableButtons()
         }
 
-        if (answeredQuestion.size == questionBank.size) {
+        if (quizViewModel.checkProgress()) {
             showProgress()
         }
     }
 
     private fun checkAnswer(userAnswer: Boolean) {
-        val correctAnswer = questionBank[currentIndex].answer
-
+        val correctAnswer = quizViewModel.currentQuestionAnswer
         val messageResId: Int
 
         if (userAnswer == correctAnswer) {
             messageResId = R.string.correct_toast
-            correctAnswers++
+            quizViewModel.correctAnswers++
```

Comment:
it would be better to increment via a method, instead directly accessing the propery